unit_tests/test_tester.py

```diff
@@ -298,9 +237,58 @@ def test(self):
         with self.subTest('SUBTEST2'):
             pass
 
-    @pytest.fixture(autouse=True)
-    def validate(self, event_system):
-        yield
 
-        assert self.event_summary == {'NORMAL': 2}
-        assert self.final_event.test_status == 'SUCCESS'
+@pytest.mark.parametrize("test_class, results, outcomes", [
+    pytest.param(TeardownFailsTest, {"passed": 1}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'ERROR': 1}",
+        "TEST_STATUS: FAILED",
+        "ERROR 0: save_email_data (silenced)"
+    ], id="TeardownFailsTest"),
+    pytest.param(SubtestsSuccessTest, {"passed": 1, "subtests": 2}, [
+        "EVENT_SUMMARY: {'NORMAL': 2}",
+        "TEST_STATUS: SUCCESS"
+    ], id="SubtestsSuccessTest"),
+    pytest.param(SuccessTest, {"passed": 1}, [
+        "EVENT_SUMMARY: {'NORMAL': 2}",
+        "TEST_STATUS: SUCCESS"
+    ], id="SuccessTest"),
+    pytest.param(TestErrorTest, {"passed": 1}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'ERROR': 1}",
+        "TEST_STATUS: FAILED",
+        "ERROR 0: Something went wrong"
+    ], id="TestErrorTest"),
+    pytest.param(SetupFailsTest, {"failed": 1}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'ERROR': 1}",
+        "TEST_STATUS: FAILED",
+        "ERROR 0: prepare_kms_host failed"
+    ], id="SetupFailsTest"),
+    pytest.param(CriticalErrorNotCaughtTest, {"passed": 1}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'CRITICAL': 1}",
+        "TEST_STATUS: FAILED",
+        "CRITICAL 0: Reason to fail",
+    ], id="CriticalErrorNotCaughtTest"),
+    pytest.param(SubtestAndTeardownFailsTest, {"failed": 3}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'ERROR': 1}",
+        "TEST_STATUS: FAILED",
+        "ERROR 0: save_email_data (silenced)",
+    ], id="SubtestAndTeardownFailsTest"),
+    pytest.param(SubtestAssertAndTeardownFailsTest, {"failed": 3}, [
+        "EVENT_SUMMARY: {'NORMAL': 2, 'ERROR': 1}",
+        "TEST_STATUS: FAILED",
+        "ERROR 0: save_email_data (silenced)",
+    ], id="SubtestAssertAndTeardownFailsTest"),
+])
+def test_tester_subclass(pytester, test_class, results, outcomes):
+    pytester.makepyfile(f"""
+        from unit_tests.conftest import *
+        from unit_tests.test_tester import {test_class.__name__}
+        {test_class.__name__}.__test__ = True
```

Comment:
maybe it worth adding a comment explaining why doing this, and not calling pytest commnadline with the class